src/Discovery/aggMDSUtils.jsx

```diff
@@ -0,0 +1,124 @@
+import { discoveryConfig, hostname } from '../localconf';
+
+const aggMDSURL = `${hostname}mds/aggregate`;
+const aggMDSDataURL = `${aggMDSURL}/metadata`;
+
+
+const retrieveFieldMapping = async (commonsName) => {
+  const url = `${aggMDSDataURL}/${commonsName}/field_to_columns`;
+  const res = await fetch(url);
+  if (res.status !== 200) {
+    throw new Error(`Request for field mapping at ${url} failed. Response: ${JSON.stringify(res, null, 2)}`);
+  }
+
+  const jsonResponse = await res.json();
+
+  return jsonResponse;
+};
+
+const retrieveCommonsInfo = async (commonsName) => {
+  const url = `${aggMDSDataURL}/${commonsName}/info`;
+  const res = await fetch(url);
+  if (res.status !== 200) {
+    throw new Error(`Request for commons info at ${url} failed. Response: ${JSON.stringify(res, null, 2)}`);
+  }
+
+  const jsonResponse = await res.json();
+
+  return jsonResponse;
+};
+
+const loadStudiesFromAggMDSRequests = async (offset, limit) => {
+  const url = `${aggMDSDataURL}?data=True&limit=${limit}&offset=${offset}`;
+
+  const res = await fetch(url);
+  if (res.status !== 200) {
+    throw new Error(`Request for study data at ${url} failed. Response: ${JSON.stringify(res, null, 2)}`);
+  }
+
+  const metadataResponse = await res.json();
+
+  // According to the API: https://petstore.swagger.io/?url=https://gist.githubusercontent.com/craigrbarnes/3ac95d4d6b5dd08d280a28ec0ae2a12d/raw/58a562c9b3a343e535849ad4085a3b27942b2b57/openapi.yaml#/Query/metadata_metadata_get
+  const commons = Object.keys(metadataResponse);
+
+  let allStudies = [];
+  for (let j = 0; j < commons.length; j += 1) {
+    const commonsName = commons[j];
+    // Now retrieve field mappings for each commons
+    // eslint-disable-next-line no-await-in-loop
+    const fieldMapping = await retrieveFieldMapping(commonsName);
+    // eslint-disable-next-line no-await-in-loop
+    const commonsInfo = await retrieveCommonsInfo(commonsName);
```

Comment:
probably better if we just break this part into another for loop so we can use `await Promise.all(...)` for each of these so we don't do await in loop